CloudyLauncher.java

```diff
@@ -266,24 +380,61 @@ private void setGameDisplayPage() {
 
     @FXML
     private void setSettingsPage(ActionEvent event) {
-        System.out.println("settings");
+        try {
+            FXMLLoader loader = new FXMLLoader(getClass().getResource(PATH_SETTINGS));
+            loader.setController(this);
+            currentStage.getScene().setRoot(loader.load());
+
+            welcomeText.setText(String.format(WELCOME_TEXT, username.getText()));
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            settingsFeedback.setText(ERROR_FXML_SETTINGS);
+            settingsButton.setDisable(true);
+        }
+
     }
 
-    private void initialiseLauncher(Stage stage) {
+    @FXML
+    private void changeTheme(ActionEvent event) {
+        String theme = ((Button) event.getSource()).getText().toLowerCase();
+        String themePath = String.format("styles/%s.css", theme);
+        currentStage.getScene().getStylesheets().clear();
+        currentStage.getScene().getStylesheets().add("styles/stylebase.css");
+        currentStage.getScene().getStylesheets().add(themePath);
+    }
+
+    private void initialiseStage() {
         try {
             FXMLLoader loader = new FXMLLoader(getClass().getResource(PATH_LAUNCHER_BASE));
             loader.setController(this);
-            stage = loader.load();
+
+            currentStage = loader.load();
+            currentStage.getScene().getStylesheets().add("styles/stylebase.css");
+            currentStage.getScene().getStylesheets().add("styles/cloudy.css");
+
+            currentStage.show();
+
+        } catch (IOException e) {
+            System.out.println("ERROR_FXML_BASE_STAGE");
+        }
+    }
+
+    private void initialiseLauncher() {
```

Comment:
Not very important, but as part of refactoring, you might want to make the functions flow from top down. So initialize stuff at the top, etc.